docs/guides/collections.rst

```diff
@@ -1,105 +1,144 @@
-  and assign this to the ``self.collection`` attribute. It **must** then assign ``self.appliance`` to
-  be equal to ``self.collection.appliance``
+Instantiating objects
+---------------------
 
-.. note:: The ``__init__()`` method could in the future become part automatic in the BaseEntity class
-          but this is a future feature and is not yet planned.
+Collection objects should be obtained via an ``IPAppliance`` object. All base/root level objects, that
+is objects which have an appliance as their parent, will be accessible via the ``IPAppliance`` objects
+``collections`` manager.
 
-.. warning:: Failure to comply with the above guidelines in the future may result in an Exception
-             being raised
+.. note:: Not all collections objects are yet available via the ``IPAppliance`` object yet.
 
-Example
--------
+See the example below which demonstrates how to obtain a Datastore collection, and then instantiate
+a ``Datastore`` object.
+
+.. code-block:: python
+
+                provider = get_crud('vsphere55')  # An example provider
 
-Below is an example of a generic object using the collection and entity relationships
+                dc = appliance.collections.datastore
+                dc.instantiate('name', provider)
+
+Filtering collections
+---------------------
+
+Some collections support filtering. This means that the base/root collection can be asked to supply a subset
+of the information if would normally return using the ``all()`` or indeed other methods. To filter a
+collection object, use the following pattern
 
 .. code-block:: python
 
-                from cfme.utils.appliance import BaseCollection, BaseEntity
+                dc = appliance.collection.datastore
+                dc_filtered = dc.filter({'provider': provider})
 
-                class ObjectCollection(BaseCollection):
-                    """An object collection"""
+Automatically generated filtered collections
+--------------------------------------------
 
-                    def __init__(self, appliance):
-                        self.appliance = appliance
+``BaseEntity`` objects have the special ability to create filtered collections. These appear, much
+like the ``collections`` attribute on the ``IPAppliance`` instances. Consider the example above where
+the ``Repository`` object is given the ``_collections`` attribute. This contains a dictionary of
+collection names, along with the collection class that should be instantiated. The collection is then
+instantiated with the following filter, like so:
 
-                    def instantiate(self, name, label):
-                        return Object(self, name, label)
+.. code-block:: python
 
-                    def create(self, name, label):
-                        run_create(name, label)
-                        return self.instantiate(name, label)
+                repo = appliance.collections.repositories.all()[0]
```

Comment:
for the semi-livecoding examples we ought to consider tested doctest blocks simply to ensure the demonstration is and stays correct

Reply:
Agreed for the future - Something to do with the dummy appliance travis run